native/src/main/java/io/ballerina/stdlib/http/api/client/actions/HttpClientAction.java

```diff
@@ -293,22 +296,66 @@ private static String[] getPathStringArray(BArray pathArray) {
 
     private static String constructQueryString(BMap params) {
         List<String> queryParams = new ArrayList<>();
+        Map<String, String> annotationValues = getQueryNameMapping(params);
         BString[] keys = (BString[]) params.getKeys();
         if (keys.length == 0) {
             return "";
         }
         for (BString key : keys) {
             Object value = params.get(key);
+            String queryName = key.getValue();
+            queryName = annotationValues.getOrDefault(queryName, queryName);
             String valueString = value.toString();
             if (value instanceof BArray) {
                 valueString = valueString.substring(1, valueString.length() - 1);
                 valueString = valueString.replace(QUOTATION_MARK, EMPTY);
             }
-            queryParams.add(key.getValue() + EQUAL_SIGN + valueString);
+            queryParams.add(queryName + EQUAL_SIGN + valueString);
         }
         return String.join(AND_SIGN, queryParams);
     }
 
+    /**
+     * This util function extracts the query name with the query annotation.
+     *
+     * @param params - Parameter map
+     * @return Map of string with overridden query param names
+     */
+    private static Map<String, String> getQueryNameMapping(BMap params) {
+        Map<String, String> annotationValues = new HashMap<>();
+        if (!(params.getType() instanceof RecordType queryRecord)) {
+            return annotationValues;
+        }
+        BMap<BString, Object> annotations = queryRecord.getAnnotations();
+
+        for (Map.Entry<BString, Object> annotation: annotations.entrySet()) {
+            BMap value = (BMap) annotation.getValue();
+            Object[] keys = value.getKeys();
+            for (Object annotRef: keys) {
+                String refRegex = "^ballerina/http:\\d+:(Query)$";
+                Pattern pattern = Pattern.compile(refRegex);
+                Matcher matcher = pattern.matcher(annotRef.toString());
+                if (matcher.find()) {
+                    extractedFieldName(annotationValues, annotation, value);
+                }
+            }
+        }
+        return annotationValues;
+    }
+
+    private static void extractedFieldName(Map<String, String> annotationValues, Map.Entry<BString, Object> annotation,
+                                           BMap value) {
+        for (Object object : value.values()) {
+            //store query parameters real name : $field$.<query name>
+            String regex = "(\\$field\\$\\.)";
+            String[] parts = Pattern.compile(regex).split(annotation.getKey().getValue());
+            String key = parts[1];
```

Comment:
```suggestion
            String fieldName = parts[1];
```

Does the field name is properly unescaped or do we need to manually do that?